kernal/drivers/x16/rtc.s

```diff
@@ -156,3 +162,67 @@ bin_to_bcd:
 @end:	cld
 	ply
 	rts
+
+; Inputs: 
+; Y = nvram offset
+; A = byte value (for write)
+;
+; Outputs:
+; A = byte value (for read)
+; C = 0: success
+; C = 1: failure
+;
+; clobbers X
+rtc_get_nvram:
+	clc
+	bra rtc_nvram
+rtc_set_nvram:
+	sec
+rtc_nvram:
+	php
+	pha
+
+	tya
+	and #$3f
```

Comment:
I guess the purpose of this line is to keep the offset within limits of the MCP7940N. 
Would it be better to actually verify that the offset is not above 0x3F, and return C=1 if the user tries to access an offset outside the RTC's capacity?

Reply:
Agreed, returning an error rather than forcing it to be in bounds is a better approach.